docs/examples/1_presentation/example_1_presentation_3_hb_getinfo_ping_linux_can.cpp

```diff
@@ -0,0 +1,443 @@
+                         PingPongState{"B", ping_pong_stats, {2000}, Priority::Nominal, {}, {}},
+                         PingPongState{"C", ping_pong_stats, {3000}, Priority::Nominal, {}, {}},
+                         PingPongState{"D", ping_pong_stats, {4000}, Priority::Nominal, {}, {}},
+                         PingPongState{"E", ping_pong_stats, {5000}, Priority::Nominal, {}, {}}};
+    //
+    const auto make_ping_request = [this, &ping_client](PingPongState& state) {
+        //
+        ++state.request.id;
+        if (print_activities_)
+        {
+            std::cout << "➡️  Client '" << state.name << "' sending  'Ping' req (ping_id=" << state.request.id
+                      << ",   to_node_id=" << remote_node_id_ << ")."
+                      << CommonHelpers::Printers::describeDurationInMs(uptime()) << std::endl;  // NOLINT
+        }
+        state.req_start = now();
+        ping_client.setPriority(state.priority);
+        auto maybe_promise = ping_client.request(state.req_start + 300ms, state.request);
+        ASSERT_THAT(maybe_promise, testing::VariantWith<PongPromise>(testing::_)) << "Failed to make 'Ping' request.";
+        state.promise.emplace(cetl::get<PongPromise>(std::move(maybe_promise)));
+    };
+    //
+    auto request_periodically_cb_ = executor_.registerCallback([&](const auto&) {
+        //
+        std::cout << "---------------\n";
+        for (auto& ping_pong_state : ping_pong_states)
+        {
+            make_ping_request(ping_pong_state);
+            ping_pong_state.promise->setCallback([&](const auto& arg) {
+                //
+                processPingPongResult(ping_pong_state, arg);
+            });
+        }
+    });
+    request_periodically_cb_.schedule(Callback::Schedule::Repeat{startup_time_ + 1s, 3s});
+
+    // 6. Publish heartbeats.
+    //
+    auto maybe_heartbeat_pub = presentation.makePublisher<uavcan::node::Heartbeat_1_0>();
+    ASSERT_THAT(maybe_heartbeat_pub, testing::VariantWith<HeartbeatPublisher>(testing::_))
+        << "Can't create 'Heartbeat' publisher.";
+    auto heartbeat_pub                 = cetl::get<HeartbeatPublisher>(std::move(maybe_heartbeat_pub));
+    auto publish_heartbeat_every_1s_cb = executor_.registerCallback([&](const auto& arg) {
+        //
+        EXPECT_THAT(heartbeat_pub.publish(arg.approx_now + 1s, makeHeartbeatMsg(arg.approx_now)),
+                    testing::Eq(cetl::nullopt));
+    });
+    //
+    constexpr auto hb_period = std::chrono::seconds{uavcan::node::Heartbeat_1_0::MAX_PUBLICATION_PERIOD};
+    publish_heartbeat_every_1s_cb.schedule(Callback::Schedule::Repeat{startup_time_ + hb_period, hb_period});
+
+    // 7. Bring up 'GetInfo' server.
+    //
+    using GetInfo_1_0 = uavcan::node::GetInfo_1_0;
+    uavcan::node::GetInfo_1_0::Response get_info_response{{1, 0}};
+    const std::string                   node_name{"org.opencyphal.Ex_1_Pres_3_HB_GetInfo_Ping_CAN"};
```

Comment:
Can we minimize usage of heapy containers please so that the embedded types feel at home here? I see that string_view is not available in C++14 so maybe plain `const char* const` would work?

Reply:
It just examples. I use also vector, map, ostream - but none of these at production libcyphal code.

Reply:
A bit cheap of me to pile on, but +1 to trying to minimize containers as most firmware don't have access to `std::vector` or `std::unordered_map`. Even though these are examples, examples are the biggest way that potential customers decide if a solution works from them so it might drive more embedded customers away. Probably fine to leave for now though, and thank you for making them!

Reply:
Hmm... even `std::vector`?! I'm surprised b/c Nunavut's code generation is based on the vector (but of course based on PMR allocator).

Reply:
I think our firmware applications typically use the C headers, which don't use `std::vector`